### PR TITLE
libsoup: depends on krb5 for Linuxbrew

### PR DESCRIPTION
**Pull request by maxim-belkin**

### CODE REVIEW

**Comment by maxim-belkin** [comment]
From a local build...
```
$ brew linkage libsoup
Homebrew libraries:
  ~/.linuxbrew/lib/libgio-2.0.so.0 (glib)
  ~/.linuxbrew/lib/libglib-2.0.so.0 (glib)
  ~/.linuxbrew/lib/libgobject-2.0.so.0 (glib)
  ~/.linuxbrew/lib/libc.so.6 (glibc)
  ~/.linuxbrew/lib/libcom_err.so.3 (krb5)
  ~/.linuxbrew/lib/libgssapi_krb5.so.2 (krb5)
  ~/.linuxbrew/lib/libk5crypto.so.3 (krb5)
  ~/.linuxbrew/lib/libkrb5.so.3 (krb5)
  ~/.linuxbrew/lib/libsoup-2.4.so.1 (libsoup)
  ~/.linuxbrew/lib/libxml2.so.2 (libxml2)
  ~/.linuxbrew/lib/libsqlite3.so.0 (sqlite)
Possible undeclared dependencies:
  glib
```

**Comment by maxim-belkin** [comment]
CircleCI:
```
./libsoup-2.58.0.x86_64_linux.bottle.tar.gz
  bottle do
    sha256 "ed4c6dd4866fe0a6ed2b5495715eaae169e0307240abad53e6a6585126c194d7" => :x86_64_linux
  end
==> brew bottle --merge --write --no-commit ./libsoup-2.58.0.x86_64_linux.bottle.json --keep-old
Error: --keep-old was passed but there are changes in:
sha256 => x86_64_linux
Error: Kernel.exit
==> libsoup
==> FAILED
```

**Comment by maxim-belkin** [comment]
```
==> brew linkage libsoup
System libraries:
  /lib/x86_64-linux-gnu/libc.so.6
Homebrew libraries:
  /home/linuxbrew/.linuxbrew/lib/libgio-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/lib/libglib-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/lib/libgobject-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/Cellar/krb5/1.14.4/lib/libgssapi_krb5.so.2 (krb5)
  /home/linuxbrew/.linuxbrew/Cellar/libsoup/2.58.0/lib/libsoup-2.4.so.1 (libsoup)
  /home/linuxbrew/.linuxbrew/lib/libxml2.so.2 (libxml2)
  /home/linuxbrew/.linuxbrew/lib/libsqlite3.so.0 (sqlite)
Possible undeclared dependencies:
  glib
```
adding `glib` to the list of dependencies

**Comment by maxim-belkin** [comment]
added `glib`. I expect CircleCI to fail again because of https://github.com/Linuxbrew/homebrew-core/pull/2561#issuecomment-297601455. But if this is the only failure -- will merge it without updating the bottle.

**Comment by maxim-belkin** [comment]
```
==> make install
............................
Error: Failure while executing: [#<Pathname:/home/linuxbrew/.linuxbrew/opt/patchelf/bin/patchelf>, "--print-soname", "/home/linuxbrew/.linuxbrew/Cellar/guile/2.2.2/lib/guile/2.2/ccache/ice-9/and-let-star.go"] 
==> FAILED
==> brew install --build-bottle --verbose libsoup
==> Installing dependencies for libsoup: guile, autogen, gnutls, sqlite, gdbm, openssl, berkeley-db@4, python, gsettings-desktop-schemas, glib-networking, libpng, freetype, expat, fontconfig, pixman, cairo, m4, bison, flex, gobject-introspection, vala, krb5
==> Installing libsoup dependency: guile
==> Downloading https://ftp.gnu.org/gnu/guile/guile-2.2.2.tar.xz
Already downloaded: /home/ubuntu/.cache/Homebrew/guile-2.2.2.tar.xz
==> Verifying guile-2.2.2.tar.xz checksum
==> ./configure --disable-dependency-tracking --prefix=/home/linuxbrew/.linuxbrew/Cellar/guile/2.2.2 --with-libreadline-prefix=/home/linuxbrew/.linuxbrew/opt/readline --with-libgmp-prefix=/home/linuxbrew/.linuxbrew/opt/gmp
..
==> make install
............................
Error: Failure while executing: [#<Pathname:/home/linuxbrew/.linuxbrew/opt/patchelf/bin/patchelf>, "--print-soname", "/home/linuxbrew/.linuxbrew/Cellar/guile/2.2.2/lib/guile/2.2/ccache/ice-9/and-let-star.go"] 
```

**Comment by sjackman** [comment]
Unless you're trying to address a particular build error that you think is related, you can generally ignore `Possible undeclared dependencies`. It's usually due to transitive dependencies. That is, `c` depends on `b` depends on `a`, and `c` does not have a `depends_on "a"`.

File: Formula/libsoup.rb [hunk 1]
```diff
@@ -21,7 +21,6 @@ class Libsoup < Formula
   unless OS.mac?
     depends_on "libxml2"
     depends_on "krb5"
-    depends_on "glib"
```
Comment (sjackman) [inline Formula/libsoup.rb]: I'd like to add code to `brew linkage` that detects unnecessary library dependencies. Debian has this feature.

**Comment by maxim-belkin** [comment]
rebased and force-pushed

**Comment by maxim-belkin** [comment]
```
./libsoup-2.58.0.x86_64_linux.bottle.tar.gz
  bottle do
    sha256 "7b01a6a1e21b321623df3aa8f3260e9965baf426d5ec7eeb8027f8f5acbf480f" => :x86_64_linux
  end
==> brew bottle --merge --write --no-commit ./libsoup-2.58.0.x86_64_linux.bottle.json --keep-old
Error: --keep-old was passed but there are changes in:
sha256 => x86_64_linux
Error: Kernel.exit
==> libsoup
==> FAILED
```

**Pull request closed without merging**